doc/src/site/sphinx/experimental/rest_api_based_client.rst

```diff
@@ -0,0 +1,58 @@
+Rest API Based Client
+---------------------
+
+Supporting dynamic allocation in Sparkling Water is a long-term issue and is implemented in
+several stages. The first stage is replacing H2O node running inside of the Spark executor
+by lightweight REST API client.
+
+The first iteration of replacing the H2O Node by REST API client is currently tested only
+under PySparkling in External Backend Mode. At this stage we do not support using PySparkling
+Algo API in this new approach. After the PySparkling Algo API is migrated to the new approach
+as well, we switch to the REST API Based client approach by default for PySparkling in external
+backend.
+
+Starting Sparkling Water with the REST API Client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable the REST API Based client, please start PySparkling with the spark option ``spark.ext.h2o.rest.api.based.client=true``.
```

Comment:
nit: `REST API Based` -> `REST-API-based`?

Reply:
Yup, like it!